Port WebExtension Manifest Content Scripts to C++

ItsJamie9494 opened the pull request:
#### a7f2ce55853c3958b9acf45748736dc5b3888f81
<pre>
Port WebExtension Manifest Content Scripts to C++
<a href="https://webkit.org/b/279801">https://webkit.org/b/279801</a>

Reviewed by Timothy Hatcher.

Create utility functions to filter and work with JSON arrays, and port the content scripts manifest keys to C++.

* Source/WebKit/Shared/Extensions/WebExtensionUtilities.cpp: Added.
(WebKit::filterObjects):
(WebKit::makeStringVector):
* Source/WebKit/Shared/Extensions/WebExtensionUtilities.h:
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIScriptingCocoa.mm:
(WebKit::WebExtensionContext::createInjectedContentForScripts):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm:
(WebKit::WebExtension::staticInjectedContents): Deleted.
(WebKit::WebExtension::hasStaticInjectedContentForURL): Deleted.
(WebKit::WebExtension::hasStaticInjectedContent): Deleted.
(WebKit::WebExtension::InjectedContentData::expandedIncludeMatchPatternStrings const): Deleted.
(WebKit::WebExtension::InjectedContentData::expandedExcludeMatchPatternStrings const): Deleted.
(WebKit::WebExtension::populateContentScriptPropertiesIfNeeded): Deleted.
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::addInjectedContent):
* Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
(WebKit::WebExtension::staticInjectedContents):
(WebKit::WebExtension::hasStaticInjectedContentForURL):
(WebKit::WebExtension::hasStaticInjectedContent):
(WebKit::WebExtension::InjectedContentData::expandedIncludeMatchPatternStrings const):
(WebKit::WebExtension::InjectedContentData::expandedExcludeMatchPatternStrings const):
(WebKit::WebExtension::populateContentScriptPropertiesIfNeeded):
* Source/WebKit/UIProcess/Extensions/WebExtension.h:

Canonical link: <a href="https://commits.webkit.org/283847@main">https://commits.webkit.org/283847@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c5afae8fab4e0c02db2b86846c86eef42a027ebf

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67508 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46887 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/20140 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71556 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18645 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54685 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18436 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/54083 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12471 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70575 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43032 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58383 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34545 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39705 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/17003 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61669 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/16128 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73255 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11467 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15439 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61526 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11502 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58451 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61588 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/14982 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9363 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2980 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10268 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42692 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43769 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44955 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43510 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
```diff
@@ -4301,7 +4302,7 @@ void effectiveAppearanceDidChange(Inspector::ExtensionAppearance appearance) ove
         NSArray<NSString *> *includeMatchPatterns = includeMatchPatternsSet.allObjects;
 
-        NSMutableSet<NSString *> *excludeMatchPatternsSet = [NSMutableSet setWithArray:injectedContentData.expandedExcludeMatchPatternStrings()];
+        NSMutableSet<NSString *> *excludeMatchPatternsSet = [NSMutableSet setWithArray:(::WebKit::wrapper(API::Array::createStringArray(injectedContentData.expandedExcludeMatchPatternStrings()))).get()];
```
```suggestion
        NSMutableSet<NSString *> *excludeMatchPatternsSet = [NSMutableSet setWithArray:createNSArray(injectedContentData.expandedExcludeMatchPatternStrings()).get()];
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        if (!scriptPaths)
+            scriptPaths = JSON::Array::create();
+
+        scriptPaths = filterObjects(*scriptPaths, [] (auto& value) {
```
```suggestion
        scriptPaths = filterObjects(*scriptPaths, [](auto& value) {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        if (!styleSheetPaths)
+            styleSheetPaths = JSON::Array::create();
+
+        styleSheetPaths = filterObjects(*styleSheetPaths, [] (auto& value) {
```
```suggestion
        styleSheetPaths = filterObjects(*styleSheetPaths, [](auto& value) {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        if (!includeGlobPatternStrings)
+            includeGlobPatternStrings = JSON::Array::create();
+
+        includeGlobPatternStrings = filterObjects(*includeGlobPatternStrings, [] (auto& value) {
```
```suggestion
        includeGlobPatternStrings = filterObjects(*includeGlobPatternStrings, [](auto& value) {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        if (!excludeGlobPatternStrings)
+            excludeGlobPatternStrings = JSON::Array::create();
+
+        excludeGlobPatternStrings = filterObjects(*excludeGlobPatternStrings, [] (auto& value) -> bool {
```
```suggestion
        excludeGlobPatternStrings = filterObjects(*excludeGlobPatternStrings, [](auto& value) -> bool {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+
+        auto styleLevel = WebCore::UserStyleLevel::Author;
+        String cssOriginString = injectedContentObject->getString(contentScriptsCSSOriginManifestKey);
+        if (!cssOriginString || (cssOriginString == contentScriptsAuthorManifestKey))
```
```suggestion
        if (!cssOriginString || cssOriginString == contentScriptsAuthorManifestKey)
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+
+        WebExtensionContentWorldType contentWorldType = WebExtensionContentWorldType::ContentScript;
+        String worldString = injectedContentObject->getString(contentScriptsWorldManifestKey);
+        if (!worldString || (runsAtString == contentScriptsIsolatedManifestKey))
```
```suggestion
        if (!worldString || runsAtString == contentScriptsIsolatedManifestKey)
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+
+    // Documentation: https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/content_scripts
+
+    auto contentScriptsManifestArray = manifestObject->getArray(contentScriptsManifestKey);
```
```suggestion
    RefPtr contentScriptsManifestArray = manifestObject->getArray(contentScriptsManifestKey);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        HashSet<Ref<WebExtensionMatchPattern>> includeMatchPatterns;
+
+        // Required. Specifies which pages the specified scripts and stylesheets will be injected into.
+        auto matchesArray = injectedContentObject->getArray(contentScriptsMatchesManifestKey);
```
```suggestion
        RefPtr matchesArray = injectedContentObject->getArray(contentScriptsMatchesManifestKey);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+            if (!matchPatternString)
+                continue;
+
+            if (auto matchPattern = WebExtensionMatchPattern::getOrCreate(matchPatternString)) {
```
```suggestion
            if (RefPtr matchPattern = WebExtensionMatchPattern::getOrCreate(matchPatternString)) {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        }
+
+        // Optional. The list of JavaScript files to be injected into matching pages. These are injected in the order they appear in this array.
+        auto scriptPaths = injectedContentObject->getArray(contentScriptsJSManifestKey);
```
```suggestion
        RefPtr scriptPaths = injectedContentObject->getArray(contentScriptsJSManifestKey);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        });
+
+        // Optional. The list of CSS files to be injected into matching pages. These are injected in the order they appear in this array, before any DOM is constructed or displayed for the page.
+        auto styleSheetPaths = injectedContentObject->getArray(contentScriptsCSSManifestKey);
```
```suggestion
        RefPtr styleSheetPaths = injectedContentObject->getArray(contentScriptsCSSManifestKey);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+
+        // Optional. Whether the script should inject into an about:blank frame where the parent or opener frame matches one of the patterns declared in matches. Defaults to false.
+        auto matchesAboutBlankBool = injectedContentObject->getBoolean(contentScriptsMatchesAboutBlankManifestKey);
+        bool matchesAboutBlank = matchesAboutBlankBool ? *matchesAboutBlankBool : false;
```
```suggestion
        bool matchesAboutBlank = injectedContentObject->getBoolean(contentScriptsMatchesAboutBlankManifestKey).value_or(false);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        HashSet<Ref<WebExtensionMatchPattern>> excludeMatchPatterns;
+
+        // Optional. Excludes pages that this content script would otherwise be injected into.
+        auto excludeMatchesArray = injectedContentObject->getArray(contentScriptsExcludeMatchesManifestKey);
```
```suggestion
        RefPtr excludeMatchesArray = injectedContentObject->getArray(contentScriptsExcludeMatchesManifestKey);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        if (!excludeMatchesArray)
+            excludeMatchesArray = JSON::Array::create();
+
+        for (auto matchPatternStringValue : *excludeMatchesArray) {
```
```suggestion
        for (RefPtr matchPatternStringValue : *excludeMatchesArray) {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+            if (matchPatternString.isEmpty())
+                continue;
+
+            if (auto matchPattern = WebExtensionMatchPattern::getOrCreate(matchPatternString)) {
```
```suggestion
            if (RefPtr matchPattern = WebExtensionMatchPattern::getOrCreate(matchPatternString)) {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        });
+
+        // Optional. Applied after matches to exclude URLs that match this glob.
+        auto excludeGlobPatternStrings = injectedContentObject->getArray(contentScriptsExcludeGlobsManifestKey);
```
```suggestion
        RefPtr excludeGlobPatternStrings = injectedContentObject->getArray(contentScriptsExcludeGlobsManifestKey);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        // topmost frame in a tab. Defaults to false, meaning that only the top frame is matched. If specified true, it will inject into all frames, even if the frame is not the topmost frame in
+        // the tab. Each frame is checked independently for URL requirements, it will not inject into child frames if the URL requirements are not met.
+        auto injectsIntoAllFramesBool = injectedContentObject->getBoolean(contentScriptsAllFramesManifestKey);
+        bool injectsIntoAllFrames = injectsIntoAllFramesBool ? *injectsIntoAllFramesBool : false;
```
```suggestion
        bool injectsIntoAllFrames = injectedContentObject->getBoolean(contentScriptsAllFramesManifestKey).value_or(false);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+
+        InjectionTime injectionTime = InjectionTime::DocumentIdle;
+        String runsAtString = injectedContentObject->getString(contentScriptsRunAtManifestKey);
+        if (!runsAtString || (runsAtString == contentScriptsDocumentIdleManifestKey))
```
```suggestion
        auto injectionTime = InjectionTime::DocumentIdle;
        auto runsAtString = injectedContentObject->getString(contentScriptsRunAtManifestKey);
        if (!runsAtString || runsAtString == contentScriptsDocumentIdleManifestKey)
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+
+        WebExtensionContentWorldType contentWorldType = WebExtensionContentWorldType::ContentScript;
+        String worldString = injectedContentObject->getString(contentScriptsWorldManifestKey);
+        if (!worldString || runsAtString == contentScriptsIsolatedManifestKey)
```
This one is important (was comparing `runsAtString` instead of `worldString`):
```suggestion
        auto contentWorldType = WebExtensionContentWorldType::ContentScript;
        auto worldString = injectedContentObject->getString(contentScriptsWorldManifestKey);
        if (!worldString || worldString == contentScriptsIsolatedManifestKey)
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+            recordError(createError(Error::InvalidContentScripts, WEB_UI_STRING("Manifest `content_scripts` entry has unknown `world` value.", "WKWebExtensionErrorInvalidContentScripts description for unknown 'world' value")));
+
+        auto styleLevel = WebCore::UserStyleLevel::Author;
+        String cssOriginString = injectedContentObject->getString(contentScriptsCSSOriginManifestKey);
```
```suggestion
        auto cssOriginString = injectedContentObject->getString(contentScriptsCSSOriginManifestKey);
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+    };
+
+    for (auto contentScriptsManifestEntry : *contentScriptsManifestArray) {
+        if (auto injectedContentObject = contentScriptsManifestEntry->asObject())
```
```suggestion
    for (RefPtr injectedContentValue : *contentScriptsManifestArray) {
        if (RefPtr injectedContentObject = contentScriptsManifestEntry->asObject())
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        }
+
+        for (auto matchPatternStringValue : *matchesArray) {
+            String matchPatternString = matchPatternStringValue->asString();
```
```suggestion
        for (RefPtr matchPatternStringValue : *matchesArray) {
            auto matchPatternString = matchPatternStringValue->asString();
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,233 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        return;
+    }
+
+    auto addInjectedContentData = [this](auto injectedContentObject) {
```
```suggestion
    auto addInjectedContentData = [this](auto& injectedContentObject) {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
```diff
@@ -4301,7 +4302,7 @@ void effectiveAppearanceDidChange(Inspector::ExtensionAppearance appearance) ove
         NSArray<NSString *> *includeMatchPatterns = includeMatchPatternsSet.allObjects;
 
-        NSMutableSet<NSString *> *excludeMatchPatternsSet = [NSMutableSet setWithArray:injectedContentData.expandedExcludeMatchPatternStrings()];
+        NSMutableSet<NSString *> *excludeMatchPatternsSet = [NSMutableSet setWithArray:createNSArray(injectedContentData.expandedExcludeMatchPatternStrings()).get()];
```
```suggestion
        auto *excludeMatchPatternsSet = [NSMutableSet setWithArray:createNSArray(injectedContentData.expandedExcludeMatchPatternStrings()).get()];
```

xeenon reviewed (changes requested):
Found a bug with `world` that needs fixed. Also saw more suggestions. Should always use `Ref` or `RefPtr` instead of `auto`. This is a new code style recommendation for WebKit to assist in smart pointer adoption audits.

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,231 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        if (!excludeMatchesArray)
+            excludeMatchesArray = JSON::Array::create();
+
+        for (RefPtr matchPatternStringValue : *excludeMatchesArray) {
```
```suggestion
        for (Ref matchPatternStringValue : *excludeMatchesArray) {
```

xeenon commented on Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
```diff
@@ -240,6 +260,231 @@ void WebExtension::populatePagePropertiesIfNeeded()
+        m_staticInjectedContents.append(WTFMove(injectedContentData));
+    };
+
+    for (auto injectedContentValue : *contentScriptsManifestArray) {
```
```suggestion
    for (Ref injectedContentValue : *contentScriptsManifestArray) {
```
I think it is `Ref`.

ItsJamie9494 replied:
`error: no viable constructor or deduction guide for deduction of template arguments of 'Ref'`

ItsJamie9494 committed:
Port WebExtension Manifest Content Scripts to C++
https://webkit.org/b/279801

Reviewed by Timothy Hatcher.

Create utility functions to filter and work with JSON arrays, and port the content scripts manifest keys to C++.

* Source/WebKit/Shared/Extensions/WebExtensionUtilities.cpp: Added.
(WebKit::filterObjects):
(WebKit::makeStringVector):
* Source/WebKit/Shared/Extensions/WebExtensionUtilities.h:
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIScriptingCocoa.mm:
(WebKit::WebExtensionContext::createInjectedContentForScripts):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm:
(WebKit::WebExtension::staticInjectedContents): Deleted.
(WebKit::WebExtension::hasStaticInjectedContentForURL): Deleted.
(WebKit::WebExtension::hasStaticInjectedContent): Deleted.
(WebKit::WebExtension::InjectedContentData::expandedIncludeMatchPatternStrings const): Deleted.
(WebKit::WebExtension::InjectedContentData::expandedExcludeMatchPatternStrings const): Deleted.
(WebKit::WebExtension::populateContentScriptPropertiesIfNeeded): Deleted.
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::addInjectedContent):
* Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
(WebKit::WebExtension::staticInjectedContents):
(WebKit::WebExtension::hasStaticInjectedContentForURL):
(WebKit::WebExtension::hasStaticInjectedContent):
(WebKit::WebExtension::InjectedContentData::expandedIncludeMatchPatternStrings const):
(WebKit::WebExtension::InjectedContentData::expandedExcludeMatchPatternStrings const):
(WebKit::WebExtension::populateContentScriptPropertiesIfNeeded):
* Source/WebKit/UIProcess/Extensions/WebExtension.h:

Canonical link: https://commits.webkit.org/283847@main
